how-to-docs/elastic-container-registry.md

```diff
@@ -0,0 +1,143 @@
+
+
+### Pulling & Pushing Images
+
+Within the ACP Kubernetes Clusters, you do not need to provide an `imagePullSecret` as was previously required for images in Artifactory. The ACP Clusters will authenticate behind-the-scenes and be able to successfully pull images from any Docker Repositories you create via the Platform Hub.
+
+The Docker Repositories section of the Platform Hub will provide a URL such as follows for the Repository you have created: `670930646103.dkr.ecr.eu-west-2.amazonaws.com/acp/hello-world-app`
+
+Now that you have locally authenticated with AWS ECR, you can pull and push images as normal:
+
+```bash
+$ docker build . -t 670930646103.dkr.ecr.eu-west-2.amazonaws.com/acp/hello-world-app:v0.0.1
+
+Sending build context to Docker daemon  32.78MB
+...
+Successfully built 882e2cadb649
+Successfully tagged 670930646103.dkr.ecr.eu-west-2.amazonaws.com/acp/hello-world-app:v0.0.1
+
+$ docker push 670930646103.dkr.ecr.eu-west-2.amazonaws.com/acp/hello-world-app:v0.0.1
+
+The push refers to repository [670930646103.dkr.ecr.eu-west-2.amazonaws.com/acp/hello-world-app]
+afbe4b47c182: Pushed
+78147c906fce: Pushed
+86177d14466d: Pushed
+f55514f6bd18: Pushed
+ce74984572d7: Pushed
+67d7e5db87ee: Pushed
+12d012372115: Pushed
+b0bb54920d03: Pushed
+835c2760f26b: Pushed
+e9bcacee1741: Pushed
+cd7100a72410: Pushed
+v0.0.1: digest: sha256:0309d2655ecef6b4181ee93edfb91f386fc2ebc7849cc88f6e7a18b0d349c35f size: 2628
+
+$ docker pull 670930646103.dkr.ecr.eu-west-2.amazonaws.com/acp/hello-world-app:v0.0.1@sha256:0309d2655ecef6b4181ee93edfb91f386fc2ebc7849cc88f6e7a18b0d349c35f
+
+sha256:0309d2655ecef6b4181ee93edfb91f386fc2ebc7849cc88f6e7a18b0d349c35f: Pulling from acp/hello-world-app
+Digest: sha256:0309d2655ecef6b4181ee93edfb91f386fc2ebc7849cc88f6e7a18b0d349c35f
+Status: Image is up to date for 670930646103.dkr.ecr.eu-west-2.amazonaws.com/acp/hello-world-app@sha256:0309d2655ecef6b4181ee93edfb91f386fc2ebc7849cc88f6e7a18b0d349c35f
+```
+
+## Managing Image Deployments via Drone CI
+
+The Docker Authorisation Token generated via the aws-cli command is only valid for 12 hours, and so this can't be used as a Drone Secret for Docker Image builds. Instead, you would need to store the IAM Credentials for a Robot Account as Drone Secrets and perform the `aws ecr get-login` + `docker login ..` step on each build.
+
+Below is an example Drone CI Pipeline, using the AWS IAM Credentials to retrieve an authorisation token for docker login to ECR:
+```yml
+pipeline:
+
+  build:
+    image: docker:18.09.0
+    environment:
+    - DOCKER_HOST=tcp://172.17.0.1:2375
+    commands:
+    - docker build -t hello-world-app:$${DRONE_COMMIT_SHA} .
+
+  ecr_push:
+    image: docker:18.09.0
+    secrets:
+    - AWS_ACCESS_KEY_ID
+    - AWS_SECRET_ACCESS_KEY
+    environment:
+    - AWS_DEFAULT_REGION=eu-west-2
+    - AWSCLI_IMAGE=quay.io/ukhomeofficedigital/aws-cli:latest
```

Comment:
Might be worth doing
`ENV DOCKER_HOST=tcp://172.17.0.1:2375` in the image so we can avoid the line below everywhere

Reply:
It's to do the `docker run` so would actually need to be in the base dind image (`image: docker:18.09.0`). Might be worth us creating one for use, will simplify builds for users a little bit.